### ISSUE TITLE
Qube Manager should remove RPM-installed templates or tell the user how

### ISSUE THREAD

**Issue by straussmaus** [QubesOS/qubes-issues#5509]
**Qubes OS version**
R4.0.1

**Affected component(s) or functionality**
Qubes Manager

**Brief summary**
Can not delete whonix-gw-14 after installing whonix-gw-15.

**To Reproduce**

_1. Go to Qubes Manager
_2. RIGHT Click on whonix-gw-14
_3. Choose "Delete qube"
_4. Enter "whonix-gw-14"
_5. See error: `VM installed by package manager: whonix-gw-14`

**Expected behavior**
The error message should indicate what to do next so I can delete this qube.

**Actual behavior**
No actionable information presented.

**Additional context**
It is a nuisance to have old whonix versions after upgrading to whonix-gw-15 so this is a non-trivial problem.  I have a constant warning that `whonix-gw-14` needs to be upgraded so the Qubes Updater requires extra steps constantly, etc.

**Solutions you've tried**
Only tried as above.

**Relevant [documentation](https://www.qubes-os.org/doc/) you've consulted**
None.  Looking at and searching https://www.qubes-os.org/doc/ does not bring up anything obvious.  If docs exist it is not clear where.

**Related, [non-duplicate](https://www.qubes-os.org/doc/reporting-bugs/#new-issues-should-not-be-duplicates-of-existing-issues) issues**
None found.

**Comment by andrewdavidwong** [comment]
You should be able to uninstall that template by issuing `sudo dnf remove qubes-template-whonix-gw-14` in a dom0 terminal. I agree that it would be an enhancement for the Qube Manager to either do this for you or tell you what to do.

**Comment by straussmaus** [comment]
@andrewdavidwong 

Thank you and this gets to a very important point.  Many many many issues come because of bad error messaging.  I do not mean to say the work is not very good but maybe a Meta-Issue is that "inline error/docs must be actionable" and this would save so many problems.  How many times do you write "did you read the docs?"  Solution: good error messaging because it is a general HIG that an error must lead to some productive end.

Big enhancement but maybe a bigger saver of time?

**Comment by marmarta** [comment]
I think this particular issue should be solved via template manager - it should have options to both remove templates installed via package and install them. And, of course, the error should mention it.
And, well, while I agree in general that "making error messages actionable" is a very important goal, there's so many of them at the moment we can only tackle them as those issues arise one by one. I'm also pretty skeptical about people reading error messages (I mean people who make GitHub tickets etc usually do, but my sad experience is that most others don't) :(

**Comment by straussmaus** [comment]
@marmarta 

> I'm also pretty skeptical about people reading error messages (I mean people who make GitHub tickets etc usually do, but my sad experience is that most others don't)

But who are you building Qubes for?  The people who take the time to read error messages (but don't see what action to take) or the others who don't?

> I think this particular issue should be solved via template manager

What is the template manager?  I have used Qubes for years and I don't know what you mean.

**Comment by marmarta** [comment]
> > I think this particular issue should be solved via template manager
> 
> What is the template manager? I have used Qubes for years and I don't know what you mean.

A fairly new tool (last year or so), qubes-template-manager from CLI or Qubes Template Manager in the menu.

**Comment by straussmaus** [comment]
@marmarta  

> I think this particular issue should be solved via template manager

Thank you.  Now I tried the qubes-template-manager and I disagree.  Because it is normally possible to delete a Qube in the Qubes Manager, it should be *always* possible to delete a Qube in the Qubes Manager (and that may sometimes include seeing an error message that says `You should be able to uninstall that template by issuing 'sudo dnf remove qubes-template-whonix-gw-14' in a dom0 terminal.`

**Comment by 0spinboson** [comment]
Coming at it from a slightly different direction, how important is it really to users who can't or won't read the docs / are insufficiently motivated to ask in the qubes user group to be able to remove repo/installer-sourced templates? Will they really miss the few GB associated with having a superfluous / outdated template installed?
I don't disagree that the error message could be slightly more informative, but even that strikes me as having fairly low priority.

**Comment by straussmaus** [comment]
@0spinboson 

> I don't disagree that the error message could be slightly more informative, but even that strikes me as having fairly low priority.

I disagree.  I think useful error messages are Priority #1.  More important than writing code.  But I am not a developer on this project so I can only apply that rule to my projects not someone else's projects.

> how important is it really to users who can't or won't read the docs / are insufficiently motivated to ask in the qubes user group to be able to remove repo/installer-sourced templates?

I don't know but that isn't really the point.  @marmarta made a point that I disagree with but let's not pigeon walk into a completely different place.  The important thing is that any user should be able to delete Template B the same way they delete Template A.  If there is something unusual about Template B there must be an explanation *in that workflow* what to do instead.

**Comment by straussmaus** [comment]
@marmarta

> A fairly new tool (last year or so), qubes-template-manager from CLI or Qubes Template Manager in the menu.

This is a great tool and I can get to it from the CLI but I have never seen it in the menus.  Am I going blind?

**Comment by straussmaus** [comment]
@andrewdavidwong 

> You should be able to uninstall that template by issuing sudo dnf remove qubes-template-whonix-gw-14 in a dom0 terminal. I agree that it would be an enhancement for the Qube Manager to either do this for you or tell you what to do.

This is still happening for whonix-gw-15 in Nov 2021.  The fix you mentioned above also still works.

**Comment by unman** [comment]
[quote]
This is a great tool and I can get to it from the CLI but I have never seen it in the menus.  Am I going blind?
[/quote]
In 4.1 it is in the "Qubes Tools" Menu category.

**Comment by andrewdavidwong** [comment]
> This is still happening for whonix-gw-15 in Nov 2021.

Since this issue is still open, I think it's still in the process of being rolled out. The last qubesos-bot update in https://github.com/QubesOS/qubes-issues/issues/5509#issuecomment-615552364 states that it's in the 4.1 testing repo.

**Comment by DemiMarie** [comment]
Is this still an issue?

**Comment by straussmaus** [comment]
@DemiMarie

I don't know.  Qubes Manager is not starting for me at the moment (https://github.com/QubesOS/qubes-issues/issues/7649)

I will test again when it is.   - although I do not know if I have a Qube to test with, so might have to wait until the next whonix update to 17
